mlir/include/mlir/Dialect/Rock/IR/MfmaInsnGroup.h

```diff
@@ -0,0 +1,124 @@
+//===- MfmaInsnGroup.h - MLIR to C++ for Rock conversion
+//---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// This file implements code selection logic for Mfma instructions.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_MFMA_INSN_GROUP_H
+#define MLIR_MFMA_INSN_GROUP_H
+
+#include <set>
+
+#include "mlir/Dialect/AMDGPU/AMDGPUDialect.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringMap.h"
+
+namespace mlir {
+namespace rock {
+
+enum class MfmaTypeId { Fp32TyId = 0, Fp16TyId, Bf16TyId, I8TyId };
+
+struct MfmaInsnInfo {
+  MfmaTypeId type;
+  int64_t mfmaNonKDim;
+  int64_t k;
+  int64_t blocksMfma;
+};
+
+struct MfmaInsnAttr {
+  int64_t mfmaNonKDim;
+  int64_t k;
+  int64_t blocksMfma;
+
+  int64_t nInputsToMfma;
+  int64_t k_base;
+  int64_t inputSpanLen;
+  int64_t inputSpansPerMfmaIn;
+
+  int64_t nOutputsOfMfma;
+  int64_t rowGroupSize;
+  int64_t rowsPerMfmaOutput;
+  int64_t blocksPerMfmaOutput;
+  int64_t rowGroupsPerBlock;
+  int64_t blocksInOutRegs;
+};
+
+class MfmaInsn {
+private:
+  static const llvm::StringMap<MfmaInsnInfo> mfmaInsnInfoMap;
+  static const llvm::StringMap<MfmaInsnAttr> mfmaInsnAttrMap;
+  static MfmaInsnAttr deriveAttr(MfmaInsnInfo info);
+  MfmaInsnAttr attr;
+
+public:
+  static FailureOr<MfmaInsn> select(StringRef mfmaInsn);
+  MfmaInsn(const MfmaInsnAttr &mfmaInsnAttr);
+
+  MfmaInsnAttr getAttr();
+  Type getArgType(Type elementType);
+  VectorType getRetType(Type elementType);
+  bool isCohereantWithK(int64_t kPack, int64_t kPerBlock);
```

Comment:
Nit: did you mean Coherent (`isCoherentWithK`)?

Reply:
Nice catch!